### ISSUE TITLE
Traction Sandbox Hyperledger Aries/AnonCreds Workshop: FAQ, Questions and Troubleshooting

### ISSUE THREAD

**Issue by swcurran** [bcgov/traction#927]
This issue is for those running the [Hyperledger Aries/AnonCreds Traction Sandbox Workshop] to see an FAQ and to ask questions.

[Hyperledger Aries/AnonCreds Traction Sandbox Workshop]: https://bit.ly/AnonCredsTraction

As we collect questions and develop answers, we'll add FAQs here. For now, there are none -- everything is working perfectly!

**Comment by jgriffithGitHub** [comment]
I was able to run Lab 1, but for Lab 2, the "Add Schema form Ledger" button was disabled, so I couldn't run any other labs. What did I miss?

**Comment by swcurran** [comment]
Hi @jgriffithGitHub — try using this:`CTD3jXaqbKyfyJh4WN7y1N:2:student id:1.0.0` — I think that should work. Please confirm and if it does, I’ll do a PR to update the docs.  We recently changed the ledger that the Traction Instance points to, hence the issue.

**Comment by jgriffithGitHub** [comment]
I can't get to a screen where I can enter a value, so it's not a problem with a specific schema. 
The button that displays that screen is disabled (see screenshot)

![AddSchemaDisabled](https://github.com/bcgov/traction/assets/7537157/edd0b7dc-0299-466e-852b-df16817e4d7d)

**Comment by swcurran** [comment]
@loneil — has something changed in the Sandbox deployment in the endorser handling?

**Comment by loneil** [comment]
Hi @jgriffithGitHub the step 4 in Lab1 ("Make your new Traction Tenant a verifiable credential issuer by:") must be completed before the ability to manage schemas and creds is available.

In the user Profile once you connect to the Endorser you should see the Issuer badge and the buttons will be available
![image](https://github.com/bcgov/traction/assets/17445138/f5f73d54-ac9a-4b77-a422-8b5655194a7c)

**Comment by jgriffithGitHub** [comment]
Thanks.

**Comment by jgriffithGitHub** [comment]
I see what I missed.. I had to click on the icon in the left-most column. I
was clicking on the words "bcovrin-test" which wasn't doing anything.

Thanks for the quick replies.

**Comment by jgriffithGitHub** [comment]
The schema in the lab worked.

H7W22uhD4ueQdGaGeiCgaM:2:student id:1.0.0

I was able to work through the labs with it.

![image](https://github.com/bcgov/traction/assets/7537157/4067d66c-8237-4541-99cb-3ccdeadb83aa)

**Comment by swcurran** [comment]
Awesome — thanks.  Closing this issue. Should probably add a note in the document about this via a PR.

**Comment by icredyRepo** [comment]
Is there a change in BCcovrin network to accept the new endorser and agents ?
I have installed local sandbox environment of TRACTION https://github.com/bcgov/traction?tab=readme-ov-file.  The Aca-py Version 1.0 was working fine. I was able to connect to Tenant to the Endorser agent and issue schemas to the bcovrin ledger. But just recently that version stopped working. When i create a new tenant and try to connect to endorser from profile option of the tenant UI  i get the message that ,request is set, but tenant is not connected to endorser. 
  To investigate further i tried to establish connection between tenant Agent and Endorser agent through there swaggers and what i found is the  connection request is sent by tenant (tenant/endorser-connection) and as a response i got a following response. According to this it seems that the accept is set as manual and not sure why is that because the Agent config file based on which endorser container was build has a auto-accept- connection request set as true.
  I have tried the same with latest version of traction where agents are upgraded to version 1.1.0 and have the same behaviour.
{
  "state": "request",
  "created_at": "2024-12-10T04:57:06.625901Z",
  "updated_at": "2024-12-10T04:57:06.650905Z",
  "connection_id": "a6b09e33-ed27-490b-93c5-ba35b5f6729c",
  "my_did": "DzTXxYKShHZxxGVEtySUbc",
  "their_did": "SVfHGCEEvEFmpBPcxgNqRR",
  "their_role": "inviter",
  "connection_protocol": "didexchange/1.0",
  "rfc23_state": "request-sent",
  "request_id": "216fd327-9598-4906-9a8e-df6ab83ead0d",
  "accept": "manual",
  "invitation_mode": "once",
  "alias": "bcovrin-test-endorser",
  "their_public_did": "SVfHGCEEvEFmpBPcxgNqRR"
}

**Comment by swcurran** [comment]
FYI @WadeBarnes and @loneil — please take a look at the [previous comment](https://github.com/bcgov/traction/issues/927#issuecomment-2547317674)

**Comment by esune** [comment]
@icredyRepo I checked out the latest main (`d422ebce8ec86149ce24b797e82bf2e29fe4e1c6`) and tested creating a tenant and connecting to an endorser: everything works as expected.
Please note that `bcovrin-test-endorser` in the list is set as "quick-connect" (little lightning bolt besides it), so the tenant UI will automatically refresh and publish a did automatically, while `bcovrin-test-endorser-1` will require you to manually refresh the view to see if the connection is active, and you will have to trigger the public did promotion as well.
![Image](https://github.com/user-attachments/assets/9db16507-2872-4e67-8f7d-00186add70e5)

If your setup does not work, please check that all your containers are running and that your ngrok tunnel is active - it becomes stale after some time.

If you still need support feel free to log a new issue - this was related to a workshop that happened a few months back and used a hosted version of Traction (https://traction-sandbox-tenant-ui.apps.silver.devops.gov.bc.ca) and not to the actual code.

**Comment by ananyachibber21** [comment]
When we are establishing connection between 2 traction agents, I noticed that the DID that is created does not have any DID method specified. However when I try using swagger API GET/resolver/resolve/{did} to resolve its DID Document, a document is resolved using the prefix did:sov but does not resolves with did:peer. The confusion is that a DID Document should not be resolved using the did:sov method as we are using for communication only.

**Comment by icredyRepo** [comment]
<img width="515" alt="Image" src="https://github.com/user-attachments/assets/6d6f68f3-6ad2-442d-bdb0-ef6a7d692140" />


> [@icredyRepo](https://github.com/icredyRepo) I checked out the latest main (`d422ebce8ec86149ce24b797e82bf2e29fe4e1c6`) and tested creating a tenant and connecting to an endorser: everything works as expected. Please note that `bcovrin-test-endorser` in the list is set as "quick-connect" (little lightning bolt besides it), so the tenant UI will automatically refresh and publish a did automatically, while `bcovrin-test-endorser-1` will require you to manually refresh the view to see if the connection is active, and you will have to trigger the public did promotion as well. ![Image](https://github.com/user-attachments/assets/9db16507-2872-4e67-8f7d-00186add70e5)
> 
> If your setup does not work, please check that all your containers are running and that your ngrok tunnel is active - it becomes stale after some time.
> 
> If you still need support feel free to log a new issue - this was related to a workshop that happened a few months back and used a hosted version of Traction (https://traction-sandbox-tenant-ui.apps.silver.devops.gov.bc.ca) and not to the actual code.


Hi




> [@icredyRepo](https://github.com/icredyRepo) I checked out the latest main (`d422ebce8ec86149ce24b797e82bf2e29fe4e1c6`) and tested creating a tenant and connecting to an endorser: everything works as expected. Please note that `bcovrin-test-endorser` in the list is set as "quick-connect" (little lightning bolt besides it), so the tenant UI will automatically refresh and publish a did automatically, while `bcovrin-test-endorser-1` will require you to manually refresh the view to see if the connection is active, and you will have to trigger the public did promotion as well. ![Image](https://github.com/user-attachments/assets/9db16507-2872-4e67-8f7d-00186add70e5)
> 
> If your setup does not work, please check that all your containers are running and that your ngrok tunnel is active - it becomes stale after some time.
> 
> If you still need support feel free to log a new issue - this was related to a workshop that happened a few months back and used a hosted version of Traction (https://traction-sandbox-tenant-ui.apps.silver.devops.gov.bc.ca) and not to the actual code.

Hi Esun, 
     I have check all the containers are up and running. Traction agent swagger is running at port 9301 i can unlock API's using change-me password. But still i am not able to connect to endorser from Profile of Traction Tenant. Following is the error i am getting. 
<img width="754" alt="Image" src="https://github.com/user-attachments/assets/aa5e5215-88ae-4505-99c6-63560fe2e19f" />
